### PR TITLE
fix: async FL algorithm to follow FedBuff in aggregation

### PR DESCRIPTION
**Pull request by jaemin-shin**
changed the aggregation process to only aggregate delta weights, scale down when agg goal is achieved, and then add the scaled delta weights to the original weights.

## Description

Please provide a meaningful description of what this change will do, or is for. Bonus points for including links to
related issues, other PRs, or technical references.

Note that by _not_ including a description, you are asking reviewers to do extra work to understand the context of this
change, which may lead to your PR taking much longer to review, or result in it not being reviewed at all.

## Type of Change

- [x] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: lib/python/flame/mode/horizontal/asyncfl/middle_aggregator.py [hunk 1]
```diff
@@ -224,8 +220,10 @@ def _aggregate_weights(self, tag: str) -> None:
         # save global weights before updating it
         self.prev_weights = self.weights
 
-        # set global weights
-        self.weights = self._agg_goal_weights
+        # set global weights, by adding scaled aggregated weights with aggregation goal
+        self.weights = self.optimizer.scale_add_agg_weights(
+            self.weights, self._agg_goal_weights, self._agg_goal
```
Comment (myungjin) [inline lib/python/flame/mode/horizontal/asyncfl/middle_aggregator.py:225]: Looks like self.weights and self.prev_weights point to the same object.
If so, we need to do a `deepcopy` here.
Can you please confirm if those two point to different objects?

**Review by myungjin — COMMENTED** [review]
left one comment

File: lib/python/flame/mode/horizontal/asyncfl/middle_aggregator.py (hunk 1 above)
Comment (jaemin-shin) [reply lib/python/flame/mode/horizontal/asyncfl/middle_aggregator.py:225]: Found these are the same; updated the code to deepcopy weights on prev_weights! Thank you very much for nice catch :)

**Review by myungjin — APPROVED** [review]
lgtm

**Pull request merged**